### PR TITLE
perf: optimize sequencer performance

### PR DESCRIPTION
**Pull request by bnoieh**
### Description
This PR optimized op-node/sequence actions to improve chain TPS 

### Rationale
When chain traffic increase,  empty block rate will increase, which limit the on-chain performance.
This PR reduced empty block rate to solve this issue

### Example
use `--sequencer.priority` to enable this feature

### Changes
1. cache payload by hash after sequencer successfully built block
2. sequencer step takes precedence over derivate step
3. immediately trigger start building next payload after sealing parent block

### CODE REVIEW

File: op-node/rollup/driver/state.go [hunk 1]
```diff
@@ -228,7 +246,15 @@ func (s *Driver) eventLoop() {
 				// This may adjust at any time based on fork-choice changes or previous errors.
 				//
 				// update sequencer time if the head changed
-				planSequencerAction()
+				delay := s.sequencer.PlanNextSequencerAction()
+				if delay == 0 {
+					if err := sequencerStep(); err != nil {
+						return
+					}
+					continue
```
Comment (bendanzhentan) [inline op-node/rollup/driver/state.go:254]: What's purpose of this `continue`? Could you be able to add some comments?

File: op-node/rollup/driver/state.go [hunk 2]
```diff
@@ -228,7 +246,15 @@ func (s *Driver) eventLoop() {
 				// This may adjust at any time based on fork-choice changes or previous errors.
 				//
 				// update sequencer time if the head changed
```
Comment (bendanzhentan) [inline op-node/rollup/driver/state.go:248]: These comments are for `planSequencerAction`. Please move these comments down to `planSequencerAction`.

File: op-node/rollup/driver/state.go [hunk 3]
```diff
@@ -243,112 +269,107 @@ func (s *Driver) eventLoop() {
...
-				// Errors are not severe enough to change/halt sequencing but should be logged and metered.
-				if err := s.network.PublishL2Payload(ctx, payload); err != nil {
-					s.log.Warn("failed to publish newly created block", "id", payload.ID(), "err", err)
-					s.metrics.RecordPublishingError()
+		default:
+			select {
+			case <-sequencerCh:
+				if err := sequencerStep(); err != nil {
+					return
 				}
```
Comment (bendanzhentan) [inline op-node/rollup/driver/state.go:280]: Consider changing to this style, which will be less diffs.

```go
		select {
		case <-sequencerCh:
			if err := sequencerStep(); err != nil {
				return
			}
		default:
		}

     select {
      ...... original select .....
     }
```

File: op-node/rollup/driver/state.go [hunk 4]
```diff
@@ -23,7 +23,7 @@ import (
 type SyncStatus = eth.SyncStatus
 
 // sealingDuration defines the expected time it takes to seal the block
-const sealingDuration = time.Millisecond * 50
+const sealingDuration = time.Millisecond * 1
```
Comment (bendanzhentan) [inline op-node/rollup/driver/state.go:26]: I don't quite understand how 1ms `sealingDuration` can "immediately trigger start building next payload after sealing parent block"

**Review by keroro520 — APPROVED** [review]

**Review by bendanzhentan — APPROVED** [review]

**Review by andyzhang2023 — APPROVED** [review]

**Review by bendanzhentan — APPROVED** [review]

**Pull request merged**